### PR TITLE
HIR-355: Adding new API while adding filter, HIR-358: deleting filters from dashboard

### PR DESCRIPTION
**Pull request by devashishsharma2302**

### CODE REVIEW

File: squealy/squealy-web/src/Components/dashboard-view/Dashboard.js [hunk 1]
```diff
@@ -314,6 +325,10 @@ export default class Dashboard extends Component {
               value={selectedFilter.apiUrl}
               onChange={(event)=>this.updateFilterDefinition('apiUrl', event.target.value)}
             />
+            <img src={AddIcon}
+                    className='add-icon'
+                   onClick={()=>this.chartApiModalVisibilityEnabler()}
```
Comment (swapnil95) [inline squealy/squealy-web/src/Components/dashboard-view/Dashboard.js]: Why is an arrow function needed over here ?

File: squealy/squealy-web/src/Components/dashboard-view/ChartApiModal.js [hunk 2]
```diff
@@ -125,9 +132,17 @@ export default class ChartApiModal extends Component {
...
+    this.setState({
+      apiDefinition: [getEmptyApiDefinition()],  
+      testData: [getEmptyTestData()]
+    })
+  }
+
 	render() {
 	const {showModal, closeChartApiModal, saveChartApi} = this.props
-	const modalContent = 
+  const modalContent =
```
Comment (swapnil95) [inline squealy/squealy-web/src/Components/dashboard-view/ChartApiModal.js]: Indentation is not correct

File: squealy/squealy-web/src/Components/dashboard-view/Dashboard.js [hunk 3]
```diff
@@ -314,6 +325,10 @@ export default class Dashboard extends Component {
               value={selectedFilter.apiUrl}
               onChange={(event)=>this.updateFilterDefinition('apiUrl', event.target.value)}
             />
+            <img src={AddIcon}
+                    className='add-icon'
```
Comment (swapnil95) [inline squealy/squealy-web/src/Components/dashboard-view/Dashboard.js]: Indentation is not correct

File: squealy/squealy-web/src/Components/dashboard-view/ChartApiModal.js (hunk 2 above)
Comment (devashishsharma2302) [inline squealy/squealy-web/src/Components/dashboard-view/ChartApiModal.js]: fixed

File: squealy/squealy-web/src/Components/dashboard-view/Dashboard.js (hunk 1 above)
Comment (devashishsharma2302) [inline squealy/squealy-web/src/Components/dashboard-view/Dashboard.js]: fixed

File: squealy/squealy-web/src/Components/dashboard-view/Dashboard.js (hunk 3 above)
Comment (devashishsharma2302) [inline squealy/squealy-web/src/Components/dashboard-view/Dashboard.js]: fixed

**Pull request merged**